tests/tck/features/bugfix/MTSafeConcurrencyVariables.feature

```diff
@@ -0,0 +1,14 @@
+# Copyright (c) 2022 vesoft inc. All rights reserved.
+#
+# This source code is licensed under Apache 2.0 License.
+Feature: Test MT-safe varaibles
+
+  Scenario: Binary plan of minus
+    # It's not stable to reproduce the bug, so we run it 100 times
```

Comment:
So, what's the real reason about the crash? not thread-safe data structure usage?

Reply:
Yes, all variable should insert to map in building stage, and executor is running on MT-environment, so if executor insert new variable will lead to condition race.

Reply:
Which line of code inserts new variabes to map in runtime?

Reply:
https://github.com/vesoft-inc/nebula/blob/419172474fc5cb3b822b640710faa6db6d38fc2e/src/common/expression/ListComprehensionExpression.cpp#L34

Reply:
> https://github.com/vesoft-inc/nebula/blob/419172474fc5cb3b822b640710faa6db6d38fc2e/src/common/expression/ListComprehensionExpression.cpp#L34

Does it leads the query `YIELD 1 AS number MINUS YIELD 2 AS number` to crash?

Reply:
https://github.com/vesoft-inc/nebula/blob/419172474fc5cb3b822b640710faa6db6d38fc2e/src/graph/validator/YieldValidator.cpp#L183

Reply:
https://github.com/vesoft-inc/nebula/blob/419172474fc5cb3b822b640710faa6db6d38fc2e/src/graph/validator/YieldValidator.cpp#L187